### PR TITLE
feat: add many-to-many relationship support

### PR DESCRIPTION
**Pull request by seapagan**
## Summary

- Add `ManyToMany[T]` descriptor for defining M2M relationships on ORM models
- Automatic junction table creation with CASCADE FK constraints, unique pair constraints, and indexes on both columns
- `ManyToManyManager` API: `add()`, `remove()`, `clear()`, `set()`, `fetch_all()`, `fetch_one()`, `count()`, `exists()`, `filter()`
- Automatic reverse accessors on target models via `ReverseManyToMany` descriptor (with pending registration for forward-referenced models)
- Custom junction table names via `through` parameter, custom reverse accessor names via `related_name`
- Self-referential `symmetrical=True` option (single row per pair; reverse accessor suppressed for self-ref)
- `ManyToManyError` and `ManyToManyIntegrityError` exception classes
- Tests for descriptors, junction tables, CRUD ops, reverse access, registry, self-ref symmetry, and edge cases

## Testing
- poe ruff
- poe mypy
- pytest

## Summary by CodeRabbit

* **New Features**
  * Many-to-Many ORM: descriptor, manager, automatic junction-table creation, reverse accessors, symmetrical relations, CRUD and query surface

* **Bug Fixes / Reliability**
  * Registry controls: reset/snapshot/restore to isolate/restore registry state

* **Utilities**
  * Table-name validation helper and new M2M-specific error types

* **Documentation**
  * Guide, API reference and navigation entries with examples

* **Tests**
  * Extensive M2M test suite, registry isolation fixtures

* **Chores**
  * Minor config formatting and planning TODOs added

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tests/test_m2m.py [hunk 1]
```diff
@@ -0,0 +1,1432 @@
...
+
+            TargetConflict.conflict_attr = object()
+
+            # Python 3.10 wraps __set_name__ errors in RuntimeError.
+            with pytest.raises(
+                (AttributeError, RuntimeError),
+                match=r"Reverse M2M accessor|__set_name__",
+            ) as exc_info:
+
+                class SourceConflict(BaseDBModel):
```
[the remark this answers is not in the document]
Comment (seapagan) [reply tests/test_m2m.py]: for testing - this will cause an exception which is what the test is checking for

File: tests/test_m2m.py [hunk 2]
```diff
@@ -0,0 +1,1432 @@
...
+            monkeypatch.setattr(
+                ManyToMany, "_get_junction_table_name", bad_junction
+            )
+
+            with pytest.raises(
+                (ValueError, RuntimeError),
+                match=r"junction table|__set_name__",
+            ) as exc_info:
+
+                class SourceBad(BaseDBModel):  # pylint: disable=unused-variable
```
[the remark this answers is not in the document]
Comment (seapagan) [reply tests/test_m2m.py]: for testing - this will cause an exception which is what the test is checking for

File: sqliter/tui/demos/orm.py [hunk 3]
```diff
@@ -3,10 +3,10 @@
 from __future__ import annotations
 
 import io
-from typing import Optional
+from typing import Any, Optional, cast
```
[the remark this answers is not in the document]
Comment (seapagan) [reply sqliter/tui/demos/orm.py]: Any is used as a string "Any" so in reality this import is not needed, however Ruff will complain otherwise so its acceptable
